src/iOS/Xamarin.iOS/ViewControllers/CategoriesViewController.cs

```
@@ -66,6 +67,30 @@ public override void ViewDidLoad()
             NavigationItem.RightBarButtonItem = new UIBarButtonItem(UIImage.FromBundle("Settings"), UIBarButtonItemStyle.Plain, ViewSettingsPage);
 
             DefinesPresentationContext = true;
+
+            CheckForKey();
+        }
+
+        private async void CheckForKey()
+        {
+            try
+            {
+                await ApiKeyManager.TrySetLocalKey();
+
+                // Set the API key using the key manager.
+                Esri.ArcGISRuntime.ArcGISRuntimeEnvironment.ApiKey = ApiKeyManager.ArcGISDeveloperApiKey;
+
+                // Check that the current API key is valid.
+                ApiKeyStatus status = await ApiKeyManager.CheckKeyValidity();
+                if (status != ApiKeyStatus.Valid)
+                {
+                    NavigationController.PushViewController(new ApiKeyPrompt(), true);
+                }
+            }
+            catch(Exception ex)
```

Is this swallowing any exceptions?